logger/logger.go

```diff
@@ -77,19 +122,27 @@ func newLogWriter(config LogConfig) io.Writer {
 	if config.Quiet {
 		wlog.SetLevel(wlog.ERROR)
 	}
+	var logWriter io.Writer
+	if logCreator, ok := loggerRegistry[config.LogTarget]; ok {
+		logWriter, _ = logCreator.createLogger(config)
+	}
+	if logWriter == nil {
+		logWriter, _ = (&telegrafLogCreator{}).createLogger(config)
+	}
 
-	var writer io.Writer
-	if config.Logfile != "" {
-		var err error
-		if writer, err = rotate.NewFileWriter(config.Logfile, config.RotationInterval.Duration, config.RotationMaxSize.Size, config.RotationMaxArchives); err != nil {
-			log.Printf("E! Unable to open %s (%s), using stderr", config.Logfile, err)
-			writer = os.Stderr
-		}
-	} else {
-		writer = os.Stderr
+	if closer, isCloser := actualLogger.(io.Closer); isCloser {
+		closer.Close()
 	}
+	log.SetOutput(logWriter)
+	actualLogger = logWriter
+
+	return logWriter
+}
 
-	telegrafLog := newTelegrafWriter(writer)
-	log.SetOutput(telegrafLog)
-	return telegrafLog
+func init() {
+	tlc := &telegrafLogCreator{}
```

Comment:
This is okay though I do feel like it is a little bit overkill.  Everything is happening within the package, and all we really need is to pass over the eventlog logger and then we could initialize, without a second factory.

Reply:
It's handy when there is a handler which is not compilable on other platforms. I first though that event log would be that case, but it turned out that kardianos lib is nicely interfaced and decouples platform dependencies. It even looks the syslog could be already solved by the `eventlog` handler, because using that lib.  If it would be that case factories are really not needed. 
Should I remove that?

Reply:
I'd probably drop it for now, we can always add back later and in fact we have some other logging stuff coming along which might change the calculus.  I'll leave it to you to decide what you think is most understandable.

Reply:
I decided to keep that. Sure, it is a bit over-engineered, but it allows smoothly dynamically adding `eventlog` provider without 'ifs'